iGCS/Models/Craft/GCSCraftNotifications.m

```diff
@@ -24,4 +25,18 @@ + (void)didNavModeChangeFromLastHeartbeat:(mavlink_heartbeat_t) lastHeartbeat
     });
 }
 
++ (void)didMavModeChangeFromLastHeartbeat:(mavlink_heartbeat_t) lastHeartbeat
+                          andNewHeartbeat:(mavlink_heartbeat_t) newHeartbeat {
+
+    if (lastHeartbeat.base_mode != 0 &&
+        (lastHeartbeat.base_mode & MAV_MODE_FLAG_SAFETY_ARMED) == (newHeartbeat.base_mode & MAV_MODE_FLAG_SAFETY_ARMED)) {
+        return;
+    }
+
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [[NSNotificationCenter defaultCenter]postNotificationName:GCSCraftNotificationsCraftMavModeDidChange
```

Comment:
Ah, now I see the what I thought was confusion with the method names...

As written, the logic above will prevent craft move change actions from being run (e.g. updating the custom name on the UI) unless there was also an armed status change.

Looks like you'll be wanting to issue two conditional notifications in this method.  1 on a general mode change (or, depending on intended usage, depending on the change to currentModeName), and the other for armedstatus changes.